### ISSUE TITLE
Leveraging is_about to structure EFO automatically

### ISSUE THREAD

**Issue by matentzn** [EBISPOT/efo#675]
The suggestion from our meeting (@dosumis, @zoependlington, @matentzn) is the following:

1. Turn "measurement is_about anatomy" links into "measurement inheres_in_part_of anatomy" (maybe we use a different relationship in the final version, but for now, this is fine). My suggestion is to use SPARQL Update for that: First write a select query that gets the above, then turn into UPDATE query and run with ROBOT.
2. Selectively define X measurement terms (where X is an anatomical concept). Ideally, you would extract a table using SPARQL Select, that looks like this:

| EFO-ID | anatomy |
| --- | --- |
| EFO:001 | UBERON:002 |

Using a query like that:

`Select ?efo ?uberon WHERE { something something EFO subclassof measurement, optional { isabout ?anatomy, label(?efo) matches "[a-z]+[ ]measurement" }` (obviously not the right syntax).
 From the resulting table, generated with ROBOT, pick a few ones that are good to go. Turn into DOSDP or ROBOT templated and generate equivalent class axioms:

`EFO:001 EquivalentTo: EFO:"measurement" and 'inheres in part of' some ?anatomy.` 

Merge into EFO, run reasoner, check ROBOT diff, see what subclasses we gain (i.e. additional structure). Cool so far, as we have not done any (or only a tiny bit) of manual labour, aka scalable.

3. Repeat the above for volume and count measurement, compiling them into either:

`EFO:volume measurement and 'inheres in part of' some EFO:001` or 
`EFO:measurement and inheres in part of some EFO:001 and measures some TBD:"volume" (pato?)`

See what happens. 

If the above works, GWAS people do not need to change workflows much or not at all, and still get beautifully structured branches.

**Comment by dosumis** [comment]
Some background on this:
 
GWAS curators  recently presented multiple examples of missing classification structure in the EFO measurement branch  - many corresponding to classification that could be leveraged from Uberon classification and partonomy but is currently manually implemented in a very spotty fashion (e.g. incomplete classification of brain measurements and various subtypes).  There was some agreement from OTAR and IMPC that this made GWAS hard to mine based on anatomical groupings.  The number of terms is not particularly large - so the required structure should be relatively easy to generate by switching axiomatisation and using patterns to generate appropriate intermediate terms, guided by use-case review.

**Comment by dosumis** [comment]
Working in test branch - but needs:

1. More complete Uberon import (maybe with materialized 'part of' - inferred from property hierarchy)
2. Addition of missing RO axiom:  inheres_in_part_of o part_of -> inheres_in_part_of

**Comment by matentzn** [comment]
Oh yes! 2. Is important -> @zoependlington do you feel comfortable with adding that role chain? You can slack me if you need advice. As this can potentially change inferences, it would be good to at least make sure nothing is unsatisfiable afterwards (i.e. run HermiT). 

Regarding 1 - @dosumis, do you think we are breaking part of chains due to our approach to removing signature that is not in the seed? Could you give an example? @zoependlington we need to fix it at our February meeting first thing.. This should not be the case.

**Comment by dosumis** [comment]
re role chain: this needs to go in RO.  

re part_of hierarchy: 

I haven't looked in detail, but one issue is subproperties of part_of used in Uberon, e.g.  contributes_to_morphology_of

**Comment by matentzn** [comment]
Check! We will look into sub-properties in our next meeting (me and @zoependlington).

**Comment by matentzn** [comment]
And check whether RO actually has this role chain.

**Comment by jamesamcl** [comment]
**Status triage (2026-08-26).** May we close this as superseded by the OBA migration?

This 2020 proposal (successor to #8) was to auto-structure EFO's measurement branch by rewriting `is_about` anatomy links to `inheres_in_part_of` and generating EQ axioms (`measurement and 'inheres in part of' some UBERON:x`) so the reasoner would build anatomical groupings. Last activity was January 2020, when the work was running in a test branch but blocked on (1) a fuller Uberon part-of import and (2) the missing RO role chain `inheres_in_part_of o part_of -> inheres_in_part_of`.

Current situation:

- The blocking RO role chain now exists and is already in EFO's import chain: `RO_0002314` ("characteristic of part of") carries the property chain `RO_0002314 o BFO_0000050 -> RO_0002314` in `imports/oba_import.owl`.
- Epic #787 was closed on 2026-08-04: *"we now use OBA for the measurement terms"*, with ~760 EFO measurement terms obsoleted in favour of OBA replacements. OBA terms come with pattern-generated logical definitions of exactly the form this ticket wanted to generate inside EFO, so anatomical classification is now inherited from OBA rather than built locally.
- Residually, `efo-edit.owl` still has ~4,960 `is_about` restrictions on ~4,200 active classes (none on obsoleted terms), but only ~327 of those have UBERON (anatomy) fillers — the case this ticket targeted; the majority point at PR/CHEBI entities. That residue is a question of continuing the OBA migration, and is already tracked by the open sibling tickets (#8, #985, #987, #988, #1553).

**Recommendation:** close as superseded by the OBA migration (#787). If maintainers still want anatomical grouping for the not-yet-migrated measurement terms, that seems better served by migrating them to OBA (or consolidating the remaining effort under one of the sibling tickets) than by reviving this in-EFO automation.

---
*This is an automated triage review by an AI agent; maintainers, please correct if anything here is wrong.*
